pkg/agent/agent_windows_test.go

```diff
@@ -15,13 +15,18 @@
 package agent
 
 import (
-	"testing"
+	"go.uber.org/mock/gomock"
+
+	winnettest "antrea.io/antrea/pkg/agent/util/winnet/testing"
 )
 
-func mockSetInterfaceMTU(t *testing.T, returnErr error) {
-	originalSetInterfaceMTU := setInterfaceMTU
-	setInterfaceMTU = func(ifaceName string, mtu int) error {
-		return returnErr
+func mockSetInterfaceMTU(controller *gomock.Controller, ift string, mtu int) func() {
+	originalWinnetUtil := winnetUtil
+	testWinnetInterface := winnettest.NewMockInterface(controller)
+	winnetUtil = testWinnetInterface
+	testWinnetInterface.EXPECT().SetNetAdapterMTU(ift, mtu).Return(nil).Times(1)
+
+	return func() {
+		winnetUtil = originalWinnetUtil
 	}
```

Comment:
> this is not very elegant, because you mock the whole winnet interface in this function, so you can only really do it once, for a given method.

You are right, this is not a general mock function, and it seems that we cannot use a general way like this:
`func mockUtilWinnet(ctrl *gomock.Controller, expectedCalls func(mockWinnet *winnettesting.MockInterfaceMockRecorder)) func() ` since the mock function is used in `agent.go` in which OS specific package like winnet should not be imported. 

I'll revert the related changes.